sw/cheri/tests/hyperram_test.cc

```diff
@@ -0,0 +1,294 @@
+    hyperram_cap_area[rand_cap_addr] = write_cap;
+
+    asm volatile("": : :"memory");
+
+    read_cap = hyperram_cap_area[rand_cap_addr];
+    read_val = *read_cap;
+
+    if (read_val != rand_val) {
+      failures++;
+    }
+  }
+
+  return failures;
+}
+
+int stripe_test(Capability<volatile uint32_t> hyperram_area, uint32_t initial_val) {
+  uint32_t failures = 0;
+  uint32_t cur_write_val = initial_val;
+
+  for (uint32_t addr = 0; addr < HyperramSize; addr++) {
+    hyperram_area[addr] = cur_write_val;
+    cur_write_val = ~cur_write_val;
+  }
+
+  uint32_t cur_expected_val = initial_val;
+
+  for (uint32_t addr = 0; addr < HyperramSize; addr++) {
+    uint32_t read_value = hyperram_area[addr];
+    if (read_value != cur_expected_val) {
+      failures++;
+    }
+
+    cur_expected_val = ~cur_expected_val;
+  }
+
+  return failures;
+}
+
+typedef void* (*test_fn_t)(uint32_t*);
+
+extern "C" test_fn_t get_hyperram_fn_ptr(uint32_t addr);
+
+void write_prog(Capability<volatile uint32_t>& hyperram_area, uint32_t addr) {
+  // Avoid use of global data (as it's currently broken in the test environment)
+  // by writing program data directly here.
+
+  // Test program, writes 0xdeadbeef to capability provided in first argument
+  // (ca0) and returns a capability pointing to the middle of the function
+  // (offset + 0xC from function start).
+  //
+  // li t0, 0xdeadbeef # Expands to two 32-bit instructions
+  // csw t0, 0(ca0)
+  // auipcc ca0, 0
+  // cret
+
+  hyperram_area[addr] = 0xdeadc2b7;
+  hyperram_area[addr + 1] = 0xeef28293;
+  hyperram_area[addr + 2] = 0x00552023;
+  hyperram_area[addr + 3] = 0x00000517;
+  hyperram_area[addr + 4] = 0x8082;
+}
+
+int execute_test(Capability<volatile uint32_t>& hyperram_area,
+    ds::xoroshiro::P64R32& prng, int iterations) {
+
+  int failures = 0;
+
+  for (int i = 0;i < iterations; ++i) {
+    uint32_t prog_addr = prng() % (HyperramSize - 5);
+
+    write_prog(hyperram_area, prog_addr);
+
+    uint32_t test_int = 0x0;
+    void*    test_ptr;
+
+    test_fn_t test_fn = get_hyperram_fn_ptr(HYPERRAM_ADDRESS + (prog_addr * 4));
+    test_ptr = test_fn(&test_int);
+
+    if (test_int != 0xdeadbeef) {
+      failures++;
+    }
+
+    if (!__builtin_cheri_tag_get(test_ptr)) {
+      failures++;
+    }
+
+    uint32_t expected_ptr_addr = HYPERRAM_ADDRESS + 0xC + (prog_addr * 4);
+    uint32_t test_ptr_addr = __builtin_cheri_address_get(test_ptr);
+
+    if (test_ptr_addr != expected_ptr_addr) {
+      failures++;
+    }
+  }
+
+  return failures;
+}
+
+
+/**
+ * C++ entry point for the loader.  This is called from assembly, with the
+ * read-write root in the first argument.
+ */
+[[noreturn]] extern "C" void rom_loader_entry(void *rwRoot)
+{
+  Capability<void> root{rwRoot};
+
+
+  // Create a bounded capability to the UART
+  Capability<volatile OpenTitanUart<>> uart =
+    root.cast<volatile OpenTitanUart<>>();
+  uart.address() = UART_ADDRESS;
+  uart.bounds()  = UART_BOUNDS;
+
+  uart->init(921600);
```

Comment:
nit: buad rate defined in `sw/common/defs.h`

```suggestion
  uart->init(BAUD_RATE);
```